File: app/src/main/java/com/futuretech/nfmovies/Activity/SplashActivity.java
package com.futuretech.nfmovies.Activity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.WindowManager;
import android.view.animation.Animation;
import android.view.animation.LinearInterpolator;
import android.view.animation.RotateAnimation;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;
import com.futuretech.nfmovies.R;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        if (!isTaskRoot()) {
            finish();
            return;
        }
        super.onCreate(savedInstanceState);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_splash);

        ImageView o = findViewById(R.id.splash_o);
        Animation rotateAnimation = new RotateAnimation(0f, 360f, Animation.RELATIVE_TO_SELF, 0.5f,
                Animation.RELATIVE_TO_SELF, 0.5f);
        LinearInterpolator rotate = new LinearInterpolator();
        rotateAnimation.setInterpolator(rotate);
        rotateAnimation.setDuration(1000);
        rotateAnimation.setRepeatCount(-1);
        rotateAnimation.setFillAfter(true);
        rotateAnimation.setStartOffset(0);
        o.setAnimation(rotateAnimation);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent();
                intent.setClass(SplashActivity.this, MainActivity.class);
                startActivity(intent);
                finish();
            }
        }, 1000 * 3
        );

    }

}

File: app/src/main/java/com/futuretech/nfmovies/Activity/SplashActivity.kt
package com.futuretech.nfmovies.Activity

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.WindowManager
import android.view.animation.Animation
import android.view.animation.LinearInterpolator
import android.view.animation.RotateAnimation
import android.widget.ImageView

import androidx.appcompat.app.AppCompatActivity
import com.futuretech.nfmovies.R

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        if (!isTaskRoot) {
            finish()
            return
        }
        super.onCreate(savedInstanceState)
        window.addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
        setContentView(R.layout.activity_splash)

        val o = findViewById<ImageView>(R.id.splash_o)
        val rotateAnimation = RotateAnimation(0f, 360f, Animation.RELATIVE_TO_SELF, 0.5f,
                Animation.RELATIVE_TO_SELF, 0.5f)
        val rotate = LinearInterpolator()
        rotateAnimation.interpolator = rotate
        rotateAnimation.duration = 1000
        rotateAnimation.repeatCount = -1
        rotateAnimation.fillAfter = true
        rotateAnimation.startOffset = 0
        o.animation = rotateAnimation

        Handler().postDelayed({
            val intent = Intent()
            intent.setClass(this@SplashActivity, MainActivity::class.java)
            startActivity(intent)
            finish()
        }, (1000 * 3).toLong()
        )

    }

}
